test/elbo.jl

```diff
@@ -8,9 +8,9 @@
     f = GP(kernel)
     fx = f(x, 0.1)
     fz = f(z)
-    q_ex = exact_variational_posterior(fz, fx, y)
+    q_ex = optimal_variational_posterior(fz, fx, y)
 
-    sva = SparseVariationalApproximation(fz, q_ex)
+    sva = SparseVariationalApproximation(Centred(), fz, q_ex)
```

Comment:
should this get a test case for NonCentred() as well ? could we maybe just loop over the two? :thinking:

Reply:
(Again, I think our present file structure is off here -- I'm not totally happy having elbo tests appear in test/sparse_variational.jl because the elbo functionality isn't defined until src/elbo.jl, so it's a bit of a confusing structure. Equally, I don't want to have to set up another non-centred `SparseVariationalApproximation`, when I've already got one lying around in `test/sparse_variational.jl`.)

Reply:
> we know it's correct via transitivity.

You forgot an "and if we didn't make any mistakes in multiple-dispatch method definitions"... :wink:

Reply:
True -- would this be avoided by different kinds of tests though?

Reply:
That was just my motivation for wanting to explicitly test all method definitions, i.e. not _just_ the Centred() one...

Reply:
Ah okay :) I'll resolve then.